### PR TITLE
enh (RT): Optimisation of SQL queries to reduce callbacks on the legacy service grid monitoring page

### PR DESCRIPTION
**Pull request by callapa**
## Description

The aim is to reduce the number of requests on the service grid monitoring page.
In a typical case, reduction:
- As admin: 40 to 10 queries
- As non admin: 89 to 25 queries

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [x] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [x] 22.04.x
- [x] 22.10.x
- [x] 23.04.x (master)

<h2> How this pull request can be tested ? </h2>

1. Deactivate the broker
2. Connect to the Centreon platform and access the http://lcentreon_ip/centreon/main.php?p=20204 page (Legacy service grid Monitoring page)
3. Copy/paste the url called by Centreon to retrieve the real time services in a new browser tab (url: centreon/include/monitoring/status/Services/xml/serviceGridXML.php?....)
5. Close the Centreon main page
6. Check the number of queries in the database: `show status like 'Queries';``
7. Refresh the tab where the serviceGridXML.php url is located.
8. Check the number of queries in the database again: show status like 'Queries'.

Do the subtraction.

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/www/class/centreonMonitoring.class.php [hunk 1]
```diff
@@ -128,53 +128,69 @@ public function getServiceStatusCount($host_name, $objXMLBG, $o, $status, $obj)
...
-            . " (case s.state when 0 then 3 when 2 then 0 when 3 then 2  when 3 then 2 else s.state END) as tri "
-            . "FROM hosts h, services s ";
+        $rq = <<<SQL
+            SELECT
+                h.name, s.description AS service_name, s.state, s.service_id,
+                (CASE s.state
+                    WHEN 0 THEN 3
+                    WHEN 2 THEN 0
+                    WHEN 3 THEN 2
+                    WHEN 3 THEN 2
```
Comment (abuathier) [inline centreon/www/class/centreonMonitoring.class.php:142]: duplicate WHEN
```suggestion
```

Comment (abuathier) [inline centreon/www/class/centreonMonitoring.class.php:142]: Duplicate WHEN
```suggestion
                    WHEN 3 THEN 2
```

File: centreon/www/class/centreonMonitoring.class.php [hunk 2]
```diff
@@ -128,53 +128,69 @@ public function getServiceStatusCount($host_name, $objXMLBG, $o, $status, $obj)
      */
```
Comment (abuathier) [inline centreon/www/class/centreonMonitoring.class.php:128]: Maybe be more precise in the phpdoc in the same time.
I cannot make a suggestion here about non modified code, but it seems legit to me to update that.
```suggestion
     * @param string $hostList
     * @param CentreonXMLBGRequest $objXMLBG
     * @param string $o
     * @param false|int $instance
     * @param false|int $hostgroups
     */
```

**Review by abuathier — APPROVED** [review]

File: centreon/www/class/centreonMonitoring.class.php (hunk 2 above)
Comment (callapa) [reply centreon/www/class/centreonMonitoring.class.php:128]: I was really lazy to do it :)

**Review by abuathier — APPROVED** [review]

**Review by a-launois — APPROVED** [review]

File: centreon/www/class/centreonMonitoring.class.php [hunk 3]
```diff
@@ -118,63 +118,76 @@ public function getServiceStatusCount($host_name, $objXMLBG, $o, $status, $obj)
...
-        if ($hostList == "") {
-            return array();
+        if ($hostList === '') {
+            return [];
         }
 
-        $rq = "SELECT h.name, s.description as service_name, s.state, s.service_id, "
-            . " (case s.state when 0 then 3 when 2 then 0 when 3 then 2  when 3 then 2 else s.state END) as tri "
-            . "FROM hosts h, services s ";
+        $rq = <<<SQL
```
Comment (adr-mo) [inline centreon/www/class/centreonMonitoring.class.php:133]: Just a small non-blocking comment, but as there is no variable to resolve in the request I guess you could use the Nowdoc’s syntax
```suggestion
        $rq = <<<'SQL'
```

**Review by adr-mo — APPROVED** [review]

**Pull request merged**